src/main/scala/org/broadinstitute/dsde/workbench/sam/directory/PostgresDirectoryDAO.scala

```diff
@@ -145,10 +158,16 @@ class PostgresDirectoryDAO(protected val dbRef: DbReference,
     runInTransaction { implicit session =>
       val g = GroupTable.syntax("g")
 
-      // foreign keys in accessInstructions and groupMember tables are set to cascade delete
-      // note: this will not remove this group from any parent groups and will throw a
-      // foreign key constraint violation error if group is still a member of any parent groups
-      samsql"delete from ${GroupTable as g} where ${g.name} = ${groupName}".update().apply()
+      Try {
+        // foreign keys in accessInstructions and groupMember tables are set to cascade delete
+        // note: this will not remove this group from any parent groups and will throw a
+        // foreign key constraint violation error if group is still a member of any parent groups
+        samsql"delete from ${GroupTable as g} where ${g.name} = ${groupName}".update().apply()
+      }.recoverWith {
+        case fkViolation: PSQLException if fkViolation.getSQLState == PSQLStateExtensions.FOREIGN_KEY_VIOLATION =>
+          Failure(new WorkbenchExceptionWithErrorReport(
+            ErrorReport(StatusCodes.Conflict, s"group ${groupName.value} cannot be deleted because it is a member of at least 1 other group")))
+      }.get
```

Comment:
Policies have groups - is it ever possible for a policy's group to be deleted under it?

Reply:
there is a foreign key constraint to prevent that